src/main.cc

```diff
@@ -1411,26 +1423,27 @@ SquidMainSafe(int argc, char **argv)
 
 /// computes name and ID for the current kid process
 static void
-ConfigureCurrentKid(const char *processName)
+ConfigureCurrentKid(const CommandLine &cmdLine)
 {
-    // kids are marked with parenthesis around their process names
-    if (processName && processName[0] == '(') {
-        if (const char *idStart = strrchr(processName, '-')) {
-            KidIdentifier = atoi(idStart + 1);
-            const size_t nameLen = idStart - (processName + 1);
-            assert(nameLen < sizeof(TheKidName));
-            xstrncpy(TheKidName, processName + 1, nameLen + 1);
-            if (!strcmp(TheKidName, "squid-coord"))
-                TheProcessKind = pkCoordinator;
-            else if (!strcmp(TheKidName, "squid"))
-                TheProcessKind = pkWorker;
-            else if (!strcmp(TheKidName, "squid-disk"))
-                TheProcessKind = pkDisker;
-            else
-                TheProcessKind = pkOther; // including coordinator
-        }
+    const char *kidParams = nullptr;
+    if (cmdLine.hasOption(optKid, &kidParams)) {
+        SBuf processName(kidParams);
+        SBuf kidId;
+        Parser::Tokenizer tok(processName);
+        tok.suffix(kidId, CharacterSet::DIGIT);
+        KidIdentifier = atoi(kidId.c_str());
```

Comment:
Please add a check that kidId is short enough not to oveflow an int. It has been provided externally so we cannot trust it to be a sane length yet before passing to atoi().